test/browser/spec/ol/layer/vector.test.js

```diff
@@ -193,6 +193,49 @@ describe('ol.layer.Vector', function () {
       });
     });
 
+    it('detects zero opacity images', function (done) {
+      const source = new VectorSource({
+        features: [
+          new Feature({
+            geometry: new Point([-1000000, 0]),
+            name: 'feature1',
+          }),
+          new Feature({
+            geometry: new Point([1000000, 0]),
+            name: 'feature2',
+          }),
+        ],
+      });
+
+      const style = new Style({
+        image: new Circle({
+          radius: 4,
+          fill: new Fill({
+            color: 'black',
+          }),
+        }),
+      });
+      style.getImage().setOpacity(0);
```

Comment:
Use an `Icon` style instead of the `Circle`. For circle and regular shape, the recommended way of adding opacity is to use the nested `Fill` and `Stroke` style.